### PR TITLE
debug for ext test failures on jenkins

### PR DESCRIPTION
**Pull request by gabemontero**
@bparees @csrwng FYI

### CODE REVIEW

**Comment by gabemontero** [comment]
[testonlyextended][extended:core(builds)]

File: test/extended/builds/dockerfile.go [hunk 1]
```diff
@@ -47,6 +47,16 @@ USER 1001
 			buildName := "origin-base-1"
 			g.By("expecting the Dockerfile build is in Complete phase")
 			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName, exutil.CheckBuildSuccessFn, exutil.CheckBuildFailedFn)
+			//debug for failures on jenkins
+			if err != nil {
+				bldOuput, derr := oc.Run("logs").Args("-f", "bc/origin-base").Output()
+				if derr == nil {
+					g.By(fmt.Sprintf("\n\n  build logs : %s\n\n", bldOuput))
```
Comment (csrwng) [inline test/extended/builds/dockerfile.go]: switch to using GinkgoWriter

Comment (gabemontero) [inline test/extended/builds/dockerfile.go]: will push update momentarily

**Comment by gabemontero** [comment]
[testonlyextended][extended:core(builds)]

**Comment by bparees** [comment]
lgtm, needs @smarterclayton sign off (only affects extended tests, no production code)

**Comment by smarterclayton** [comment]
Approved

On Apr 19, 2016, at 10:31 PM, Ben Parees <notifications@github.com> wrote:

lgtm, needs @smarterclayton <https://github.com/smarterclayton> sign off
(only affects extended tests, no production code)

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/8566#issuecomment-212217759>

**Comment by bparees** [comment]
[merge]

Ben Parees | OpenShift

**Comment by bparees** [comment]
[merge]

**Comment by gabemontero** [comment]
@bparees @csrwng I've pushed an expanded use of dumping logs on error as well as invocation of `df` and `du`.

**Comment by gabemontero** [comment]
[testonlyextended][extended:core(builds)]

**Comment by bparees** [comment]
[merge]

**Comment by gabemontero** [comment]
@bparees @csrwng FYI - in the last extended run on this pull, 9 of the 10 failures were the failed registry push / disk full variety.  That said, the `df` and `du` debug that was added in the extended tests and jenkins job post-build action were not useful, since they captured the state of the os instance the tests were running in.  I realized last night that we need those command executed *within* the registry pod/container.

I'll create a new pull that does either an `oc rsh` or `oc exec` into registry pod/container.

**Comment by bparees** [comment]
the df information from the ec2 instance should be useful.  the du information might not if your path assumed a path within the container.  But what did the df data show?  was the ec2 instance disk full?

**Comment by gabemontero** [comment]
I ended up not getting any du information because the "/registry" path was not present (that is the path complained about in the registry as having the disk quota exceeded, and looking at the `openshift admin registry` help, that is the default volume path).

The df from the ec2 instance did not show any fs as being remotely full.  The examples both from the ext test and the jenkins post build action were consistent throughout.  Here is an example:

Filesystem                                      1K-blocks    Used Available Use% Mounted on
/dev/xvda2                                       26202092 5365056  20837036  21% /
devtmpfs                                          1827044       0   1827044   0% /dev
tmpfs                                             1844812       0   1844812   0% /dev/shm
tmpfs                                             1844812   16864   1827948   1% /run
tmpfs                                             1844812       0   1844812   0% /sys/fs/cgroup
/dev/mapper/docker--vg-openshift--xfs--vol--dir   1038336  295188    743148  29% /mnt/openshift-xfs-vol-dir
tmpfs                                              368964       0    368964   0% /run/user/1000

I think we need a view inside the registry container.  If the df, etc. stats in there, and some examination of the /registry path show the disk is *NOT* full, then its time to pull in the registry owners and see what is up.  Or what am I missing ?

Also, upon further review, I now believe all the failures from this last run were of this space / disk quota issue.

**Comment by bparees** [comment]
well, by default the docker registry uses an empty dir volume which means it should be using the host disk w/ no limit.  however i do see that when i rsh into the docker registry pod, df shows a different available space than what my host disk shows.  i'm not sure what to make of that, so i guess it's worth gathering a df from the registry pod and see.  if it's full, we'll have to figure out why that emptydir volume isn't getting access to the full host storage.

**Comment by gabemontero** [comment]
Cool ... I was going to run that exact experiment as soon as my `make release` finished, so you saved me some time :-)

So ok, I'll start down the path of gathering that in real time from the extended tests when a build fails.

**Comment by gabemontero** [comment]
Also, I experimented a bit with prunning builds/images but didn't have much luck and did not include anything in this pull.  Based on what unfolds, we'll see if it is worth (re)pursuing that.

**Pull request merged**